### PR TITLE
Smaller update to core model

### PR DESCRIPTION
**Pull request by ddavidov**
- delete by key method added

### CODE REVIEW

File: zoolanders/Framework/Model/Database.php [hunk 1]
```diff
@@ -421,4 +421,20 @@ public function find($key){
...
+    /**
+     * Delete record by key
+     *
+     * @param $key
+     *
+     * @return bool
+     */
+    public function delete($key){
+
+        $this->query->delete()
```
Comment (Skullbock) [inline zoolanders/Framework/Model/Database.php]: i would prefer:

```
$query = $this->database->getQuery(true);
```

so we know it's always a clean one

File: zoolanders/Framework/Request/Request.php [hunk 2]
```diff
@@ -15,6 +16,31 @@
 class Request extends \JInput
 {
     /**
+     * @var array|false Request headers
```
Comment (Skullbock) [inline zoolanders/Framework/Request/Request.php]: Also add a "getHeaders()" method please can be useful :)

File: zoolanders/Framework/Request/Request.php [hunk 3]
```diff
@@ -15,6 +16,31 @@
...
+        // Capture HTTP Request headers:
+        $headers = function_exists('getallheaders') ? getallheaders() : [];
+        $this->headers = new Data($headers);
+
+        $source = null;
+
+        // Parse raw json data:
+        if(in_array($this->headers->get('Content-Type'), ['application/json', 'application/x-www-form-urlencoded'])){
+            $json = json_decode(@file_get_contents('php://input'), true);
+            $source = array_merge($_REQUEST, (array)$json);
```
Comment (Skullbock) [inline zoolanders/Framework/Request/Request.php]: do this only if json :) parsing json can be null
Also, juse parse json if it's application/json, or if there is a variable in the url called "format" = json ;)
I would place this code after construction of the parent, and set the data if not already present ;)

File: zoolanders/Framework/Model/Database.php (hunk 1 above)
Comment (ddavidov) [inline zoolanders/Framework/Model/Database.php]: ok

File: zoolanders/Framework/Request/Request.php (hunk 2 above)
Comment (ddavidov) [inline zoolanders/Framework/Request/Request.php]: Ok

File: zoolanders/Framework/Request/Request.php (hunk 3 above)
Comment (ddavidov) [inline zoolanders/Framework/Request/Request.php]: Ah, got it will do, thanks ;)

File: zoolanders/Framework/Request/Request.php [hunk 4]
```diff
@@ -15,6 +16,58 @@
...
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        // Capture HTTP Request headers:
+        $this->getHeaders();
+
+        // Parse raw json data:
+        if(($this->headers->get('Content-Type') == 'application/json') || (isset($_REQUEST['format']) && ($_REQUEST['format'] == 'json'))){
```
Comment (Skullbock) [inline zoolanders/Framework/Request/Request.php]: You can use $this->getCmd('format', false)

**Pull request merged**